x-pack/plugins/security_solution/common/detection_engine/rule_monitoring/api/detection_engine_health/get_cluster_health_schemas.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import * as t from 'io-ts';
+import type { IsoDateString } from '@kbn/securitysolution-io-ts-types';
+import type { HealthInterval } from '../../model/detection_engine_health/health_interval';
+import { HealthIntervalParameters } from '../../model/detection_engine_health/health_interval';
+import type { HealthTimings } from '../../model/detection_engine_health/health_metadata';
+import type {
+  ClusterHealthParameters,
+  ClusterHealthSnapshot,
+} from '../../model/detection_engine_health/cluster_health';
+
+export type GetClusterHealthRequestBody = t.TypeOf<typeof GetClusterHealthRequestBody>;
+export const GetClusterHealthRequestBody = t.exact(
```

Comment:
I'm curious what `t.exact(t.partial(...))` gives here. What if just `t.partial(...)` used?

Afaik `GetClusterHealthRequestBody` is passed to `buildRouteValidation()` where `exactCheck()` is called under the hood. As we have custom logic to check for extra fields in `exactCheck()` it seems that `t.exact()` isn't needed here.

Reply:
@maximpn Without `t.exact` the API successfully accepts any parameters that are not in the request schema, for example:

```json
{
	"random_parameter": 42
}
```

So looks like `buildRouteValidation` doesn't work as expected.

You can pull the PR and try this locally.

Reply:
@banderror thank you for checking it, it's clear now. I just wanted to make sure it doesn't work without `t.exact()`. `exactCheck` has a comment btw

```
NOTE: You MUST use t.exact(...) for this to operate correctly as your schema needs to remove additional keys before the compare
```

So I don't think it makes sense to create a bug ticket it's rather a way how `io-ts` works.